docs/android/v2/advanced-features/hls-timed-metadata.mdx

```diff
@@ -0,0 +1,75 @@
+---
+ title: HLS Timed Metadata
+ nav: 8.05
+---
+
+HLS Timed Metadata feature helps you synchronise certain events with the HLS stream. This can be useful for showing interactive quizzes / product overlays etc.
+
+## Requirements
+
+- 100ms Android SDK version 2.5.6 or higher
+- Active HLS stream
+
+## Sending HLS Timed Metadata
+
+To add HLS timed metadata cue to the currently running HLS stream use `setHlsSessionMetadata` API like this:
+
+<Tabs id="send-hmsTimedMetadata" items={['Kotlin', 'Java']} />
+
+<Tab id='send-hmsTimedMetadata-0'>
+
+```kotlin
+val metadata = HMSHLSTimedMetadata(
+                        payload  = payloadText,
+                        duration = durationText
+                    )
+
+hmsSDK.setHlsSessionMetadata(arrayListOf(hmsHLSTimedMetadata), object : HMSActionResultListener {
+        override fun onError(error: HMSException) {
+            Log.d(TAG, "Unable to send metadata")
+        }
+
+        override fun onSuccess() {
+           Log.d(TAG, "hls metadata sent")
+        }
+})
+```
+
+</Tab>
+
+<Tab id='send-hmsTimedMetadata-1'>
+
+```java
+
+HMSHLSTimedMetadata metaDataModel = new HMSHLSTimedMetadata(payload,duration,metaData);
+ArrayList<HMSHLSTimedMetadata> metadataModelList = new ArrayList<HMSHLSTimedMetadata>(metaDataModel);
+metadataModelList.add(metaDataModel);
+hmsSDK.setHlsSessionMetadata(metadataModelList, new HMSActionResultListener() {
+    @Override
+    public void onSuccess() {
+        Log.d(TAG, "hls metadata sending failed");
+    }
+
+    @Override
+    public void onError(@NonNull HMSException e) {
+        Log.d(TAG, "hls metadata sent successfully");
+    }
+});
+
+```
+</Tab>
+
+## Receiving HLS Timed Metadata
+
+To receive metadata on the HLS player side use `HlsMetadataHandler` class from sample app.
+
+```kotlin
```

Comment:
Let's document how to sync metadata with player time as well.

Reply:
@jashan1498 provide a code snippet here

Reply:
resolved